tools/docker/scripts/ci_entrypoint.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -e
+
+# author: Ole Schuett
+
+REPORT=/workspace/report.txt
+echo -n "StartDate: " > $REPORT
+date --utc --rfc-3339=seconds >> $REPORT
+
+# Get cp2k sources.
+if [ -n "${GIT_REF}" ]; then
+    echo -e "\n========== Fetching Git Commit =========="
+    cd /workspace/cp2k
+    git fetch origin "${GIT_BRANCH}"
+    git checkout "${GIT_REF}"
+    git submodule update --init --recursive
+    git --no-pager log -1 --pretty='%nCommitSHA: %H%nCommitTime: %ci%nCommitAuthor: %an%nCommitSubject: %s%n' | tee -a $REPORT
+
+elif [ -d  /mnt/cp2k ]; then
+    echo -e "\n========== Copying Changed Files =========="
+    #TODO: Exclusion rules could also come from .gitigore
+    rsync --exclude="*~"                      \
+          --exclude=".*/"                     \
+          --exclude="*.pyc"                   \
+          --exclude=/obj/                     \
+          --exclude=/lib/                     \
+          --exclude=/exe/                     \
+          --exclude=/regtesting/              \
+          --exclude=/tools/toolchain/         \
+          --executability                     \
+          --ignore-times                      \
+          --update                            \
+          --verbose                           \
+          --recursive                         \
+          --checksum                          \
+          /mnt/cp2k/  /workspace/cp2k/
+else
+    echo "Neighter GIT_REF nor /mnt/cp2k found - aborting."
+    exit 255
+fi
+
+
+# Update toolchain.
+if [ -d /opt/cp2k-toolchain ]; then
+    echo -e "\n========== Updating Toolchain =========="
+    cd /opt/cp2k-toolchain
+    rsync --exclude="*~"          \
+          --exclude=".*/"         \
+          --exclude="*.pyc"       \
+          --exclude="/build"      \
+          --exclude="/install"    \
+          --executability         \
+          --ignore-times          \
+          --update                \
+          --verbose               \
+          --recursive             \
+          --checksum              \
+          /workspace/cp2k/tools/toolchain/  /opt/cp2k-toolchain/
```

Comment:
dito. You might want to factor out common rsync arguments or the rsync call itself